52.md

```diff
@@ -0,0 +1,212 @@
+NIP-52
+======
+
+Calendar Events
+---------------
+
+`draft` `optional` `author:tyiu`
+
+This specification defines calendar events representing an occurrence at a specific moment or between moments. These calendar events are replaceable and referenceable per [NIP-33](33.md) and deletable per [NIP-09](09.md).
+
+Unlike the term `calendar event` specific to this NIP, the term `event` is used broadly in all the NIPs to describe any Nostr event. The distinction is being made here to discern between the two terms.
+
+## Calendar Events
+
+There are two types of calendar events represented by different kinds: date-based and time-based calendar events. Calendar events are not required to be part of a [calendar](#calendar).
+
+### Date-Based Calendar Event
+
+This kind of calendar event starts on a date and ends before a different date in the future. Its use is appropriate for all-day or multi-day events where time and time zone hold no significance. e.g., anniversary, public holidays, vacation days.
+
+#### Format
+
+The format uses a parameterized replaceable event kind `31922`.
+
+The `.content` of these events is optional and should be a detailed description of the calendar event.
+
+The list of tags are as follows:
+* `d` (required) universally unique identifier (UUID). Generated by the client creating the calendar event.
+* `name` (required) name of the calendar event
+* `start` (required) inclusive start date in ISO 8601 format (YYYY-MM-DD). Must be less than `end`, if it exists.
+* `end` (optional) exclusive end date in ISO 8601 format (YYYY-MM-DD). If omitted, the calendar event ends on the same date as `start`.
+* `location` (optional) location of the calendar event. e.g. address, GPS coordinates, meeting room name, link to video call
+* `g` (optional) [geohash](https://en.wikipedia.org/wiki/Geohash) to associate calendar event with a searchable physical location as suggested as an example by [NIP-12](12.md)
+* `p` (optional, repeated) 32-bytes hex pubkey of a participant, optional recommended relay URL, and participant's role in the meeting
+* `t` (optional, repeated) hashtag to categorize calendar event
+* `r` (optional, repeated) references / links to web pages, documents, video calls, recorded videos, etc.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <Unix timestamp in seconds>,
+  "kind": "31922",
+  "content": "<description of calendar event>",
+  "tags": [
+    ["d", "<UUID>"],
+
+    ["name", "<name of calendar event>"],
+
+    // Dates
+    ["start", "<YYYY-MM-DD>"],
+    ["end", "<YYYY-MM-DD>"],
+
+    ["start_tzid", "<IANA Time Zone Database identifier>"],
```

Comment:
`start_tzid` and `end_tzid` got copy-pasta'd from time-based calendar events into date-based calendar events. I'm going to remove them because time zones do not make sense for date-based calendar events.

Reply:
Removed in 4042ab14cc77c833e56cbf9929e0b767d006de58